### PR TITLE
Android File Saver - remove MemoryStream

### LINKED ISSUE #1605: [BUG] Unable to Save Multi GB Files using FileSaver on Android
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

I am unable to save 2GB+ files/streams using FileSaver on Android.  The attempt returns an error message of "Stream too long" even when there is enough space to write the file.  It also leaves behind a 0 byte file in the target directory.  I think this would be possible if the Android implementation wrote the contents of the stream in chunks rather than attempting to load it into an array in memory as it currently does.

### Expected Behavior

I would expect FileSaver to be able to handle a file that large when there is sufficient space on the Android instance (such as in the Windows Subsystem for Android.)  This would also eliminate the 0 byte file.

### Steps To Reproduce

1.  Open the solution from the reproduction directory.  Make sure is target an Android instance that has space to write 4 GB+ such as the Windows Subsystem for Android and run the project.
2.  Click the Button and select a target directory such as Download.
3. Note that the call to FileSaver fails with the message "Stream too long."  Check the contents of the directory you selected and notice the 0 byte file left behind.

### Link to public reproduction project repository

https://github.com/DavidJSnyder/NETMAUIBigFileSaveExample

### Environment

```markdown
- .NET MAUI CommunityToolkit: 7.0.1
- OS: Windows 11 22H2 Build 22621.2861
- .NET MAUI: 8.0.3/8.0.100
- Windows Subsystem for Android: 2310.4000.2.0
```

### Anything else?

I previously opened a discussion for this issue at https://github.com/CommunityToolkit/Maui/discussions/1554 and was asked to open a bug report.

### PR DESCRIPTION
**Pull request by VladislavAntonyuk**
### Description of Change ###

 

 ### Linked Issues ###
 

 - Fixes #1605

 ### PR Checklist ###
 
 - [ ] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [ ] Has tests (if omitted, state reason in description)
 - [ ] Has samples (if omitted, state reason in description)
 - [ ] Rebased on top of `main` at time of PR
 - [ ] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pulls 

 ### Additional information ###

### CODE REVIEW

File: src/CommunityToolkit.Maui.Core/Essentials/FileSaver/FileSaverImplementation.android.cs [hunk 1]
```diff
@@ -84,11 +84,34 @@ static AndroidUri EnsurePhysicalPath(AndroidUri? uri)
...
+			long totalRead = 0;
+			while ((bytesRead = await stream.ReadAsync(buffer, cancellationToken).ConfigureAwait(false)) > 0)
+			{
+				await fileOutputStream.WriteAsync(buffer, 0, bytesRead).WaitAsync(cancellationToken).ConfigureAwait(false);
+				totalRead += bytesRead;
+			}
+
+			if (fileOutputStream.Channel is not null)
+			{
+				await fileOutputStream.Channel.TruncateAsync(totalRead).WaitAsync(cancellationToken).ConfigureAwait(false);
```
Comment (bijington) [inline src/CommunityToolkit.Maui.Core/Essentials/FileSaver/FileSaverImplementation.android.cs:108]: What is the purpose of truncating here? Is it to ensure that we have flushed everything out to disk?

**Review by bijington — APPROVED** [review]
Looks good to me! Just a minor question that likely just helps me understand something better :)

File: src/CommunityToolkit.Maui.Core/Essentials/FileSaver/FileSaverImplementation.android.cs (hunk 1 above)
Comment (VladislavAntonyuk) [reply src/CommunityToolkit.Maui.Core/Essentials/FileSaver/FileSaverImplementation.android.cs:108]: Yes. 
Assume you have a file size 1000 bytes. Then you try overwriting it with the 500 bytes. Without truncate the file size remains 1000 bytes

Comment (bijington) [reply src/CommunityToolkit.Maui.Core/Essentials/FileSaver/FileSaverImplementation.android.cs:108]: Thanks, I hadn't considered the scenario where we are overwriting an existing file.

**Review by pictos — APPROVED** [review]
LGTM!

Just double checking, did you try to save a large file?

**Comment by VladislavAntonyuk** [comment]
it was verified with a 1 GB file on the emulator with 1 GB RAM. https://downloadtestfile.com/

**Pull request merged**